python/dtype.cpp

```diff
@@ -63,43 +66,40 @@ scipp::core::DType scipp_dtype(const py::object &type) {
 // Awful handwritten parser. But std::regex does not support string_view and
 // you have to go through c-strings in order to extract the unit scale.
 [[nodiscard]] scipp::units::Unit
-parse_datetime_dtype(const std::string_view dtype_name) {
-  if (const auto length = std::size(dtype_name); length == 13 || length == 14) {
-    if (dtype_name.substr(0, 11) == "datetime64[" &&
-        dtype_name[length - 2] == 's' && dtype_name.back() == ']') {
-      if (length == 13) {
-        // no scale given
-        return scipp::units::s;
-      }
-      if (dtype_name[11] == 'n') {
-        return scipp::units::ns;
-      } else if (dtype_name[11] == 'u') {
-        return scipp::units::us;
-      } else if (dtype_name[11] == 'm') {
-        static const auto ms = units::Unit("ms");
-        return ms;
-      }
-      throw std::invalid_argument(
-          std::string("Unsupported unit in datetime: ") + dtype_name[11] + "s");
-    }
-  } else if (length == 10) {
-    if (dtype_name == "datetime64") {
-      return scipp::units::dimensionless;
-    }
+parse_datetime_dtype(const std::string &dtype_name) {
```

Comment:
You can now maybe remove aweful handwritten parser from the comment line above :)

Reply:
Right. Still don't think it is pretty, though :-)